chain/cosmos/cosmos_chain.go

```diff
@@ -141,8 +140,48 @@ func (c *CosmosChain) SendFunds(ctx context.Context, keyName string, amount ibc.
 }
 
 // Implements Chain interface
-func (c *CosmosChain) SendIBCTransfer(ctx context.Context, channelID, keyName string, amount ibc.WalletAmount, timeout *ibc.IBCTimeout) (string, error) {
-	return c.getFullNode().SendIBCTransfer(ctx, channelID, keyName, amount, timeout)
+func (c *CosmosChain) SendIBCTransfer(ctx context.Context, channelID, keyName string, amount ibc.WalletAmount, timeout *ibc.IBCTimeout) (tx ibc.Tx, _ error) {
+	txHash, err := c.getFullNode().SendIBCTransfer(ctx, channelID, keyName, amount, timeout)
+	if err != nil {
+		return tx, fmt.Errorf("send ibc transfer: %w", err)
+	}
+	txResp, err := c.getTransaction(ctx, txHash)
+	if err != nil {
+		return tx, fmt.Errorf("failed to get transaction %s: %w", txHash, err)
+	}
+	tx.Height = uint64(txResp.Height)
+	tx.TxHash = txHash
+	// In cosmos, user is charged for entire gas requested, not the actual gas used.
+	tx.GasSpent = txResp.GasWanted
+
+	const evType = "send_packet"
+	events := txResp.Events
+
+	tx.Packet.SourcePort = tendermint.AttributeValue(events, evType, "packet_src_port")
+	tx.Packet.SourceChannel = tendermint.AttributeValue(events, evType, "packet_src_channel")
+	tx.Packet.DestPort = tendermint.AttributeValue(events, evType, "packet_dst_port")
+	tx.Packet.DestChannel = tendermint.AttributeValue(events, evType, "packet_dst_channel")
+	tx.Packet.TimeoutHeight = tendermint.AttributeValue(events, evType, "packet_timeout_height")
+	tx.Packet.Data = []byte(tendermint.AttributeValue(events, evType, "packet_data"))
+
+	var (
+		seq       = tendermint.AttributeValue(events, evType, "packet_sequence")
+		timeoutTs = tendermint.AttributeValue(events, evType, "packet_timeout_timestamp")
+	)
+
+	seqNum, err := strconv.Atoi(seq)
+	if err != nil {
+		return tx, fmt.Errorf("invalid packet sequence from events %s: %w", seq, err)
+	}
+	tx.Packet.Sequence = uint64(seqNum)
+
+	timeoutNano, err := strconv.ParseInt(timeoutTs, 10, 64)
+	if err != nil {
+		return tx, fmt.Errorf("invalid packet timestamp timeout %s: %w", timeoutTs, err)
+	}
+	tx.Packet.TimeoutTimestamp = time.Duration(timeoutNano)
+
+	return tx, nil
```

Comment:
This forces the implementation to be more specific. Is it too laborous for the implementer? 

I think it's fair to say if you send an IBC transfer, a packet must be created. 

Later in the test we call `Validate()` to ensure the data is well-formed. We can then make more granular assertions on the `ibc.Packet` for example.

Reply:
seems reasonable to me!